### ISSUE TITLE
Input: Ontology

### ISSUE THREAD

**Issue by AndreaMussmann** [OpenReqEU/dependency-detection#1]
The dependency-detection service requires an ontology as input in addition to the JSON data file. I have found and read the OpenReqOntology.pdf document as well as the OpenReqJSONschema.JSON file.
Could you please clarify: 
- Is the OpenReqJSONschema.JSON file the template from which I should create the ontology for dependency detection?
- If so,  which parts of the OpenReqJSONschema.JSON file are relevant for the dependency detection ontology file.
- Could you provide an example ontology file for the dependency-detection service?

Thank you

**Comment by jnerlich** [comment]
@quim-motger Any update here?

**Comment by quim-motger** [comment]
Hi @AndreaMussmann ,

I address your questions here:

- The OpenReqJSONschema.JSON is the JSON schema of the JSON required to be sent on the body of the request. Basically for this component it is required to send a list of requirements and the project those requirements belong to. 

- Not exactly. As stated before, OpenReqJSONschema.JSON defines the body schema for the requirements that need to be send. The ontology file is a separate file in RDF/XML language which includes the references and potential relations between dependent requirements. 

- I uploaded an ontology file example at: https://github.com/OpenReqEU/dependency-detection/blob/master/src/test/java/com/gessi/dependency_detection/openreq-rail-small.owl 

To summarize, for the dependency detection process it is necessary to send two entities:

- **JSON body** containing the list of requirements and the project they belong to using the JSON OpenReq schema. Example here: https://github.com/OpenReqEU/dependency-detection/blob/master/src/test/java/com/gessi/dependency_detection/test_dependencyDetection.json

- **Ontology file** as the one attached as example before

Hope that helps to better understand the component.

**Comment by AndreaMussmann** [comment]
Hi @quim-motger ,

Thank you for the clarification! 

However, I have another question/problem regarding the service: when I try to call the service (using curl or via the website at https://api.openreq.eu/#/services/dependency-detection) I run into a 502 Bad Gateway exception. 
Is this a server issue or do I have to look for problems on my side?

Thanks.

**Comment by quim-motger** [comment]
Hi @AndreaMussmann ,

It seems like a server issue. I'm reporting this to the ENG team and I will let you know as soon as it is fixed.

**Comment by dfucci** [comment]
@AndreaMussmann have you tried to run this locally?

**Comment by quim-motger** [comment]
Hi,

the 502 Bad Gateway issue has been solved. Calls at https://api.openreq.eu/dependency-detection/swagger-ui.html should work as expected now.

Let me know if I can help you with anything else.

Best regards,

Quim Motger

**Comment by dfucci** [comment]
Thank you @quim-motger 
Closing this issue.
